bin/map_tools/outliers_process.py

```diff
@@ -0,0 +1,130 @@
+
+    # Parsing the compositional outliers (CO) file
+    # There is one line per flanking repeat. That means two lines per CO
+    # contig_1	62056	62067	contig_1_001:62056-67629|direct_1|GGCGCGCGGCC|gc:3.67|kmer:2.63|combined:3.67|conf:1.000
+    # contig_1	67618	67629	contig_1_001:62056-67629|direct_2|GGCGCGCGGCC|gc:3.67|kmer:2.63|combined:3.67|conf:1.000
+
+    with open(comp_bed, "r") as input_table:
+        lines = (line for line in input_table if not line.startswith("#"))
+        for line1, line2 in zip(lines, lines):
+            
+            # Collecting repeats info
+            contig, repeat1_start, repeat1_end, co_info1 = line1.rstrip().split("\t")
+            coord1 = (int(repeat1_start), int(repeat1_end))
+            co_id, repeat_type1, repeat_seq1, gc, kmer, score, conf = co_info1.split('|')
+            metainfo1 = [ repeat_type1.replace('direct_1','DR').replace('inverted_1','IR'), repeat_seq1, coord1 ]
+
+            contig, repeat2_start, repeat2_end, co_info2 = line2.rstrip().split("\t")
+            coord2 = (int(repeat2_start), int(repeat2_end))
+            co_id, repeat_type2, repeat_seq2, gc, kmer, score, conf = co_info2.split('|')
+            metainfo2 = [ repeat_type2.replace('direct_2','DR').replace('inverted_2','IR'), repeat_seq2, coord2 ]
+
+            # Collecting CO location info
+            co_contig = co_id.split('_')
+            co_contig.pop(-1)
+            co_contig = '_'.join(co_contig)
+            co_start = int(co_id.split(':')[1].split('-')[0])
+            co_end = int(co_id.split(':')[1].split('-')[1])
+            co_coord = (co_start, co_end)
+            co_len = co_end - co_start
+            co_range = range(co_start, co_end + 1)
+            co_attributes = ';'.join(["mobile_element_type=compositional_outlier", gc.replace(':','='), kmer.replace(':','='), conf.replace(':','=')]) 
+            new_co_id = co_contig + '|compositional_outlier:' + str(co_start) + ':' + str(co_end)
+
+            # Removing CO overlapping rnas
+            # RNA flag turn to 1 when one RNA is fully overlapping a CO
+            rna_flag = 0
+            if co_contig in rnas_coord:
+                for rna_coord_pair in rnas_coord[co_contig]:
+                    rna_start = rna_coord_pair[0]
+                    rna_end = rna_coord_pair[1]
+                    rna_len = rna_end - rna_start
+                    rna_range = range(rna_start, rna_end + 1)
+                    intersection = len(list(set(rna_range) & set(co_range)))
+                    if intersection > 0:
+                        rna_cov = float(intersection) / float(rna_len)
+                        if rna_cov == RNA_COV_THRESHOLD:
+                            rna_flag = 1
+                            print(co_id + " removed due to RNAs genes")
+
+            # If no RNAs in CO then evaluate overlapping with integrated MGEs
+            # MGE flag turn to 1 when an MGE overlaps >= 75% of CO length
+            if rna_flag == 0:
```

Comment:
if you turn `rna_flag` into a boolean then you can just do `if not rna_flag`

Reply:
done